docs/guides/errors.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Dealing with errors
+---
+
+# Dealing with errors
+
+There are multiple different types of errors in GraphQL and each can be handled differently.
+
+In this guide we will outline the different types of errors that you will encounter when building a GraphQL server.
+
+**Note**: By default Strawberry will log all execution errors to a `strawberry.execution` logger: [https://strawberry.rocks/docs/types/schema#handling-execution-errors](../types/schema#handling-execution-errors).
+
+## GraphQL validation errors
+
+GraphQL is strongly typed and so Strawberry validates all queries before executing them. If a query is invalid it isn’t executed and instead the response contains an `errors` list:
+
+```graphql+response
+{
+  hi
+}
+---
+{
+  "data": null,
+  "errors": [
+    {
+      "message": "Cannot query field 'hi' on type 'Query'.",
+      "locations": [
+        {
+          "line": 2,
+          "column": 3
+        }
+      ],
+      "path": null
+    }
+  ]
+}
+```
+
+Each error has a message, line, column and path to help you identify what part of the query caused the error.
+
+The validation rules are part of the GraphQL specification and built into Strawberry, so there’s not really a way to customize this behavior, except to pass `validate_queries=False` when executing a query, which skips validation altogether: [https://strawberry.rocks/docs/types/schema#execute-async](../types/schema#execute-async)
+
+## GraphQL type errors
+
+When a query is executed each field must resolve to the correct type. For example non-null fields cannot return None.
+
+```python
+import strawberry
+
+@strawberry.type
+class Query:
+    @strawberry.field
+    def hello() -> str:
+        return None
+
+schema = strawberry.Schema(query=Query)
+```
+
+```graphql+response
+{
+  hello
+}
+---
+{
+  "data": null,
+  "errors": [
+    {
+      "message": "Cannot return null for non-nullable field Query.hello.",
+      "locations": [
+        {
+          "line": 2,
+          "column": 3
+        }
+      ],
+      "path": [
+        "hello"
+      ]
+    }
+  ]
+}
+```
+
+Each error has a message, line, column and path to help you identify what part of the query caused the error.
+
+## Unhandled execution errors
+
+Sometimes a resolver will throw an unexpected error due to a programming error or an invalid assumption. When this happens Strawberry catches the error and exposes it in the top level `errors` field in the response.
+
+```python
+import strawberry
+
+@strawberry.type
+class User:
+    name: str
+
+@strawberry.type
+class Query:
+    @strawberry.field
+    def user() -> User:
+        raise Exception("Can't find user")
+
+schema = strawberry.Schema(query=Query)
+```
+
+```graphql+response
+{
+  user {
+    name
+  }
+}
+---
+{
+  "data": null,
+  "errors": [
+    {
+      "message": "Can't find user",
+      "locations": [
+        {
+          "line": 3,
```

Comment:
@jkimbo shouldn't this be `"line": 2`?

![image](https://user-images.githubusercontent.com/2043821/115118074-8f579500-9f6f-11eb-834f-4a381f3373dc.png)

Reply:
Well spotted! I'll change it.

Reply:
https://github.com/strawberry-graphql/strawberry/pull/860